### ISSUE TITLE
First attempt to use crate failed.

### ISSUE THREAD

**Issue by johnoneil** [Eolu/const-gen#5]
```
   Compiling const-gen v0.4.4
error[E0658]: arbitrary expressions in key-value attributes are unstable
 --> /home/joneil/.cargo/registry/src/github.com-1ecc6299db9ec823/const-gen-0.4.4/src/lib.rs:1:10
  |
1 | #![doc = include_str!("../README.md")]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #78835 <https://github.com/rust-lang/rust/issues/78835> for more information

error: aborting due to previous error
```

Our project is configured as a workspace, so that could be related. Don't have time to debug this so do what you will with this issue.

**Comment by Eolu** [comment]
This has been fixed in 0.4.5. In case anyone is wondering, when including a file as a crate doc make sure you make it relative to CARGO_MANIFEST_DIR like so:

```rust
#![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/README.md"))]
```

Edit: This is not truly the fix. See the below post.

**Comment by Eolu** [comment]
After [making a post](https://www.reddit.com/r/rust/comments/qhb4jb/a_quick_psa_on_doc_inclusion_via_include_str/) about this, I determined that there was not actually an issue here at all. You're using an outdated version of the rust compiler (pre 1.54.0), and you could potentially run into this issue pulling any valid crate using this version or later. `arbitrary expressions in key-value attributes are unstable` is not an issue in the latest versions of the rust compiler, as this has been stabilized.
